ci: allow usage of merge queues

onerandomusername opened the pull request:
## Summary

<!-- What is this pull request for? Does it fix any issues? -->
https://github.com/DisnakeDev/disnake/queue/master
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge-group
## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

shiftinv commented:
Seems like this would depend on #819 (?), that's the reason auto-merge doesn't quite work sometimes atm.

shiftinv commented:
A couple other things:
- The `if:`s would need to be updated, otherwise almost none of the jobs will run (this may or may not be resolved in #819, should take care of that first).
- We should specify `types: [checks_requested]` as shown [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group).

onerandomusername commented:
> A couple other things:
> 
>     * The `if:`s would need to be updated, otherwise almost none of the jobs will run (this may or may not be resolved in [[meta] issues with required CI checks #819](https://github.com/DisnakeDev/disnake/issues/819), should take care of that first).
> 
>     * We should specify `types: [checks_requested]` as shown [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group).

Rebased onto #851 and updated the triggers as needed.
